### PR TITLE
Return multiple errors

### PR DESCRIPTION
**Pull request by jendakol**
Return all errors instead of only the first one.  
Doesn't use exceptions internally.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,9 +68,9 @@ case class CaseClassA(
   fieldGpb: CaseClassB,
   fieldGpbOption: Option[CaseClassB],
   fieldGpbOptionEmpty: Option[CaseClassB],
-  fieldStringsList: immutable.Seq[String],
-  fieldOptionIntegersList: Option[List[Int]],
-  fieldOptionIntegersEmptyList: Option[List[Int]]
+  fieldStrings: immutable.Seq[String],
+  fieldOptionIntegers: Option[List[Int]],
```
Comment (jakubjanecek) [inline README.md]: What's the meaning of `Option[List[Int]]`?

Comment (jendakol) [inline README.md]: By using this, you require non-empty list. If the list is empty, you get `None`.

Comment (jakubjanecek) [inline README.md]: Do you really think it would be nice to use like this? I would probably understand if it meant that the underlying GPB had the list actually set (hasList returns true) but otherwise I would rather have an empty list as an indication of "nothing".

Comment (jendakol) [inline README.md]: Unfortunately repeated fields does not have `has` method...  
This is only an option, you can still have only `List[Int]` and it will just become an empty list.  
Problem with repeated fields in GPB is you have no way how to know whether someone disn't set the field or he had nothing to add there (= put there an empty list).

Comment (jakubjanecek) [inline README.md]: OK, I understand better now. It kind of makes sense however in my opinion it is not clear because type as `Option[List[_]]` are not common in Scala (usually empty list is used as `None` in this case) and it just makes the library unnecessarily more complex - it's trying to do more than it needs to. I would vote for removing this feature. You are not offering full validation of the GPB anyway... you still have to check whether the values makes sense and emptiness of a list is one of them IMHO.

Comment (jakubjanecek) [inline README.md]: This will be changed by another MR.
